packages/tools/src/utilities/segmentation/InterpolationManager/deleteRelatedAnnotations.ts

```diff
@@ -0,0 +1,120 @@
+import { triggerEvent } from '@cornerstonejs/core';
+import * as annotationStateManagement from '../../../stateManagement/annotation';
+import interpolate from '../../contours/interpolation/interpolate';
+import type { InterpolationViewportData } from '../../../types/InterpolationTypes';
+import getInterpolationDataCollection from '../../contours/interpolation/getInterpolationDataCollection';
+import { AnnotationInterpolationRemovedEventDetail } from '../../../types/EventTypes';
+import EventTypes from '../../../enums/Events';
+
+/**
+ * deleteRelatedAnnotations - Delete the same interpolation uid on deleting one of them.
+ *
+ * @param eventData - Object.
+ * @returns null
+ */
+export default function deleteRelatedAnnotations(
+  viewportData: InterpolationViewportData
+) {
+  const { annotation } = viewportData;
+  const interpolationAnnotations = getInterpolationDataCollection(
+    viewportData,
+    [{ key: 'interpolationUID', value: viewportData.interpolationUID }],
+    true
+  );
+
+  const referencedSliceIndex = annotation.metadata.referencedSliceIndex;
+  let nextNonInterpolatedIndex = null;
+  for (
+    let i = referencedSliceIndex + 1, j = referencedSliceIndex - 1;
+    i < interpolationAnnotations.length, j >= 0;
+    i++, j--
+  ) {
+    let interpolationAnnIndex = interpolationAnnotations.findIndex(
+      (ann) => ann.metadata.referencedSliceIndex === i
+    );
```

Comment:
Let's consider creating a precached map for these slice indices. I encountered a similar performance issue in the segmentation import, and addressing it with a map outside of the for loop resulted in a significant 3x performance improvement.